diff --git a/Makefile b/Makefile
index bddcf6f..edb08b9 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@ all:
 	go build -tags "$(BUILDTAGS)" -o runc .
 
 static:
-	CGO_ENABLED=1 go build -tags "$(BUILDTAGS) cgo static_build" -ldflags "-w -extldflags -static" -o runc .
+	CGO_ENABLED=1 go build -tags "$(BUILDTAGS) cgo static_build" -ldflags "-extldflags -static" -o runc .
 
 vet:
 	go get golang.org/x/tools/cmd/vet
