diff --git a/Makefile b/Makefile
index ddd8fa6..b5df7d9 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ endif
 	go build -ldflags "-X main.gitCommit=${COMMIT}" -tags "$(BUILDTAGS)" -o runc .
 
 static:
-	CGO_ENABLED=1 go build -tags "$(BUILDTAGS) cgo static_build" -ldflags "-w -extldflags -static -X main.gitCommit=${COMMIT}" -o runc .
+	CGO_ENABLED=1 go build -tags "$(BUILDTAGS) cgo static_build" -ldflags "-extldflags -static -X main.gitCommit=${COMMIT}" -o runc .
 
 lint:
 	go vet ./...
