diff --git a/build/root/Makefile b/build/root/Makefile
index 64fd4aa4..d60f35f3 100644
--- a/build/root/Makefile
+++ b/build/root/Makefile
@@ -353,6 +353,29 @@ define RELEASE_HELP_INFO
 #   make release-in-a-container
 endef
 .PHONY: release release-in-a-container
+
+RELEASE_VER := $(shell git tag)
+KUBE_IMAGE := ssst0n3/kubernetes_dev:${RELEASE_VER}
+KUBE_BASE_IMAGE_REGISTRY ?= registry.k8s.io/build-image
+KUBE_CROSS_IMAGE ?= $(KUBE_BASE_IMAGE_REGISTRY)/kube-cross
+KUBE_CROSS_VERSION ?= $(shell cat build/build-image/cross/VERSION)
+
+.PHONY: image
+image:
+	@set -e; \
+	build_image="$(KUBE_CROSS_IMAGE):$(KUBE_CROSS_VERSION)"; \
+	docker pull "$$build_image"; \
+	docker tag "$$build_image" "$(KUBE_IMAGE)"; \
+	echo "Built image registry: $(KUBE_IMAGE)"
+
+.PHONY: shell
+shell:
+	@docker run -ti \
+		--user $(shell id -u):$(shell id -g) \
+		-e HOME=/tmp \
+		-w /go/src/k8s.io/kubernetes \
+		-v $(shell pwd):/go/src/k8s.io/kubernetes \
+		${KUBE_IMAGE} bash -c "git config --global --add safe.directory /go/src/k8s.io/kubernetes && exec bash"
 ifeq ($(PRINT_HELP),y)
 release release-in-a-container:
 	echo "$$RELEASE_HELP_INFO"
