diff --git a/Dockerfile b/Dockerfile
index 5c65470..d5d75f8 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,10 +1,13 @@
 FROM golang:1.12-stretch
 
+RUN sed -i s/deb.debian.org/archive.debian.org/g /etc/apt/sources.list
+RUN sed -i s/security.debian.org/archive.debian.org/g /etc/apt/sources.list
+
 RUN dpkg --add-architecture armel \
     && dpkg --add-architecture armhf \
     && dpkg --add-architecture arm64 \
     && dpkg --add-architecture ppc64el \
-    && apt-get update && apt-get install -y \
+    && apt-get update || apt-get install -y \
     build-essential \
     curl \
     sudo \
