diff --git a/Dockerfile b/Dockerfile
index 8ddf636..9a2eb09 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -4,6 +4,7 @@ RUN echo 'deb http://archive.debian.org/debian jessie-backports main' > /etc/apt
 RUN sed -i s/httpredir.debian.org/archive.debian.org/g /etc/apt/sources.list
 RUN sed -i s/security.debian.org/archive.debian.org/g /etc/apt/sources.list
 RUN apt-get update -o Acquire::Check-Valid-Until=false -o Acquire::AllowInsecureRepositories=true || apt-get install -y --force-yes \
+    libapparmor-dev \
     build-essential \
     curl \
     iptables \
