diff --git a/Makefile b/Makefile
index c822bf7..ec04bf1 100644
--- a/Makefile
+++ b/Makefile
@@ -86,9 +86,6 @@ fmt:
 lint:
 	@hack/validate-lint
 
-shell: dbuild
-	$(DOCKER_RUN) bash
-
 test: validate install bundles-rootfs
 	go test -bench=. -v $(shell go list ./... | grep -v /vendor | grep -v /integration-test ) -runtime=$(RUNTIME)
 ifneq ($(wildcard /.dockerenv), )
@@ -107,3 +104,12 @@ validate: fmt lint
 
 uninstall:
 	$(foreach file,containerd containerd-shim ctr,rm /usr/local/bin/$(file);)
+
+RELEASE_VER := $(shell git tag)
+CONTAINERD_IMAGE := ssst0n3/containerd_dev:${RELEASE_VER}
+
+image:
+	docker buildx build -f Dockerfile -t ${CONTAINERD_IMAGE} .
+
+shell:
+	docker run -ti -v $(shell pwd):/go/src/github.com/containerd/containerd ${CONTAINERD_IMAGE} bash
