diff --git a/build/root/Makefile b/build/root/Makefile
index f463cd96..28029f2c 100644
--- a/build/root/Makefile
+++ b/build/root/Makefile
@@ -97,6 +97,14 @@ all:
	hack/make-rules/build.sh $(WHAT)
 endif
 
+.PHONY: debug
+debug: DBG = 1
+debug: KUBE_DEBUG_FORCE_STATIC = 1
+debug: KUBE_STATIC_OVERRIDES += kubelet
+debug: GOFLAGS += -buildvcs=false
+debug:
+	hack/make-rules/build.sh $(WHAT)
+
 define GINKGO_HELP_INFO
 # Build ginkgo
 #
