diff --git a/Makefile.debug b/Makefile.debug
new file mode 100644
index 0000000..0c06bd8
--- /dev/null
+++ b/Makefile.debug
@@ -0,0 +1,27 @@
+RELEASE_VER := $(shell git tag)
+# from .github/workflows/release.yml
+UBUNTU_VERSION := 22.04
+GO_VERSION := 1.23.8
+PLATFORM := linux/amd64
+GATEWAY := 172.21.16.1
+PREFIX_LEN := 12
+BUILD_ARGS=--build-arg GATEWAY=${GATEWAY} --build-arg PREFIX_LEN=${PREFIX_LEN}
+CONTAINERD_IMAGE := ssst0n3/containerd_dev:${RELEASE_VER}
+
+.PHONY: image
+image:
+    # reference: https://github.com/containerd/containerd/blob/v2.1.0/.github/workflows/release.yml#L118
+	docker buildx build --build-arg ${RELEASE_VER} --build-arg UBUNTU_VERSION=${UBUNTU_VERSION} --build-arg GO_VERSION=${GO_VERSION} ${BUILD_ARGS} -f .github/workflows/release/Dockerfile --platform=${PLATFORM} --target shell -t ${CONTAINERD_IMAGE} .
+
+.PHONY: shell
+shell:
+	docker run -ti -v $(PWD):/go/src/github.com/containerd/containerd ${CONTAINERD_IMAGE} bash
+
+.PHONY: debug
+debug:
+	mv debug/00-debug-flag.go debug/00-debug-flag.go.bak || echo ""
+	make GODEBUG=1 GO_BUILD_FLAGS='-gcflags=all="-N -l"'
+	for f in ${COMMANDS}; do mv bin/$${f} bin/$${f}-${RELEASE_VER}-debug; done
+	mv debug/00-debug-flag.go.bak debug/00-debug-flag.go || echo ""
+	make GODEBUG=1 GO_BUILD_FLAGS='-gcflags=all="-N -l"'
+	for f in ${COMMANDS}; do mv bin/$${f} bin/$${f}-${RELEASE_VER}-debug-flag; done
diff --git a/debug/00-debug-flag.go b/debug/00-debug-flag.go
new file mode 100644
index 0000000..51129e4
--- /dev/null
+++ b/debug/00-debug-flag.go
@@ -0,0 +1,28 @@
+package debug
+
+import (
+	"os"
+	"strings"
+	"time"
+)
+
+func hasCmd() bool {
+	args := os.Args[1:]
+	l := len(args)
+	if l <= 1 {
+		return true
+	}
+	return strings.HasPrefix(args[l-2], "-")
+}
+
+func init() {
+	if !hasCmd() {
+		var flag bool
+		for {
+			if flag {
+				break
+			}
+			time.Sleep(time.Second)
+		}
+	}
+}
\ No newline at end of file
diff --git a/debug/placeholder.go b/debug/placeholder.go
new file mode 100644
index 0000000..9577941
--- /dev/null
+++ b/debug/placeholder.go
@@ -0,0 +1 @@
+package debug
\ No newline at end of file
