diff --git a/.github/workflows/release/Dockerfile b/.github/workflows/release/Dockerfile
index 3de5295..a659254 100644
--- a/.github/workflows/release/Dockerfile
+++ b/.github/workflows/release/Dockerfile
@@ -42,6 +42,7 @@ ARG PREFIX_LEN
 FROM ${TARGETOS} AS shell
 WORKDIR /go/src/github.com/containerd/containerd
 COPY --from=go /usr/local/go /usr/local/go
+RUN git config --global --add safe.directory /go/src/github.com/containerd/containerd
 
 FROM ${TARGETOS} AS target
 WORKDIR /go/src/github.com/containerd/containerd
