diff --git a/Makefile b/Makefile
index c1cc845..a5572a5 100644
--- a/Makefile
+++ b/Makefile
@@ -322,3 +322,12 @@ image:
 .PHONY: shell
 shell:
 	docker run -ti -v $(shell pwd):/go/src/github.com/containerd/containerd ${CONTAINERD_IMAGE} bash
+
+.PHONY: debug
+debug:
+	mv debug/00-debug-flag.go debug/00-debug-flag.go.bak || echo ""
+	make GODEBUG=1
+	for f in ${COMMANDS}; do mv bin/$${f} bin/$${f}-${RELEASE_VER}-debug; done
+	mv debug/00-debug-flag.go.bak debug/00-debug-flag.go || echo ""
+	make GODEBUG=1
+	for f in ${COMMANDS}; do mv bin/$${f} bin/$${f}-${RELEASE_VER}-debug-flag; done
diff --git a/cmd/containerd-shim-runc-v1/main.go b/cmd/containerd-shim-runc-v1/main.go
index 1b1b106..611a809 100644
--- a/cmd/containerd-shim-runc-v1/main.go
+++ b/cmd/containerd-shim-runc-v1/main.go
@@ -18,6 +18,10 @@
 
 package main
 
+import (
+    _ "github.com/containerd/containerd/debug"
+)
+
 import (
 	v1 "github.com/containerd/containerd/runtime/v2/runc/v1"
 	"github.com/containerd/containerd/runtime/v2/shim"
diff --git a/cmd/containerd-shim-runc-v2/main.go b/cmd/containerd-shim-runc-v2/main.go
index 4f5d804..22c73ba 100644
--- a/cmd/containerd-shim-runc-v2/main.go
+++ b/cmd/containerd-shim-runc-v2/main.go
@@ -18,6 +18,10 @@
 
 package main
 
+import (
+	_ "github.com/containerd/containerd/debug"
+)
+
 import (
 	v2 "github.com/containerd/containerd/runtime/v2/runc/v2"
 	"github.com/containerd/containerd/runtime/v2/shim"
diff --git a/cmd/containerd-shim/main_unix.go b/cmd/containerd-shim/main_unix.go
index 49f16e6..b3bd724 100644
--- a/cmd/containerd-shim/main_unix.go
+++ b/cmd/containerd-shim/main_unix.go
@@ -18,6 +18,10 @@
 
 package main
 
+import (
+	_ "github.com/containerd/containerd/debug"
+)
+
 import (
 	"bytes"
 	"context"
diff --git a/cmd/containerd-stress/main.go b/cmd/containerd-stress/main.go
index 9ef058b..aa4382d 100644
--- a/cmd/containerd-stress/main.go
+++ b/cmd/containerd-stress/main.go
@@ -16,6 +16,10 @@
 
 package main
 
+import (
+	_ "github.com/containerd/containerd/debug"
+)
+
 import (
 	"context"
 	"encoding/json"
diff --git a/cmd/containerd/main.go b/cmd/containerd/main.go
index 10bde45..1a2316d 100644
--- a/cmd/containerd/main.go
+++ b/cmd/containerd/main.go
@@ -16,6 +16,10 @@
 
 package main
 
+import (
+	_ "github.com/containerd/containerd/debug"
+)
+
 import (
 	"fmt"
 	"os"
diff --git a/cmd/ctr/main.go b/cmd/ctr/main.go
index cf72de2..71e9e08 100644
--- a/cmd/ctr/main.go
+++ b/cmd/ctr/main.go
@@ -16,6 +16,10 @@
 
 package main
 
+import (
+	_ "github.com/containerd/containerd/debug"
+)
+
 import (
 	"fmt"
 	"os"
diff --git a/debug/00-debug-flag.go b/debug/00-debug-flag.go
new file mode 100644
index 0000000..51129e4
--- /dev/null
+++ b/debug/00-debug-flag.go
@@ -0,0 +1,28 @@
+package debug
+
+import (
+	"os"
+	"strings"
+	"time"
+)
+
+func hasCmd() bool {
+	args := os.Args[1:]
+	l := len(args)
+	if l <= 1 {
+		return true
+	}
+	return strings.HasPrefix(args[l-2], "-")
+}
+
+func init() {
+	if !hasCmd() {
+		var flag bool
+		for {
+			if flag {
+				break
+			}
+			time.Sleep(time.Second)
+		}
+	}
+}
\ No newline at end of file
diff --git a/debug/placeholder.go b/debug/placeholder.go
new file mode 100644
index 0000000..9577941
--- /dev/null
+++ b/debug/placeholder.go
@@ -0,0 +1 @@
+package debug
\ No newline at end of file
