diff --git a/Dockerfile b/Dockerfile
index f5ce90c..cc76adb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -15,12 +15,11 @@ RUN echo 'deb https://download.opensuse.org/repositories/devel:/tools:/criu/Debi
     && dpkg --add-architecture ppc64el \
     && apt-get update \
     && apt-get install -y --no-install-recommends \
+        gperf \
         build-essential \
         criu \
         crossbuild-essential-arm64 \
-        crossbuild-essential-armel \
         crossbuild-essential-armhf \
-        crossbuild-essential-ppc64el \
         curl \
         gawk \
         gcc \
@@ -29,13 +28,10 @@ RUN echo 'deb https://download.opensuse.org/repositories/devel:/tools:/criu/Debi
         kmod \
         libseccomp-dev \
         libseccomp-dev:arm64 \
-        libseccomp-dev:armel \
         libseccomp-dev:armhf \
-        libseccomp-dev:ppc64el \
         libseccomp2 \
         pkg-config \
         python-minimal \
         sudo \
         uidmap \
     && apt-get clean \