diff --git a/Makefile b/Makefile
index 0a15fd9..10b918e 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ MANDIR := $(PREFIX)/share/man
 
 GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
 GIT_BRANCH_CLEAN := $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g")
-RUNC_IMAGE := runc_dev$(if $(GIT_BRANCH_CLEAN),:$(GIT_BRANCH_CLEAN))
+RUNC_IMAGE := ssst0n3/runc_dev:v1.2.3
 PROJECT := github.com/opencontainers/runc
 EXTRA_BUILDTAGS :=
 BUILDTAGS := seccomp urfave_cli_no_docs
@@ -185,7 +185,7 @@ localrootlessintegration: all
 	tests/rootless.sh
 
 .PHONY: shell
-shell: runcimage
+shell:
 	$(CONTAINER_ENGINE) run $(CONTAINER_ENGINE_RUN_FLAGS) \
 		-ti --privileged --rm \
 		-v $(CURDIR):/go/src/$(PROJECT) \
